src/main/java/com/osp/dragon/service/ArticleService.java

```diff
@@ -14,18 +16,19 @@
 @Service
 public class ArticleService {
 
-    private final Map<String, Article> articlesMap;
 
-    public ArticleService() {
-        this.articlesMap = initializeArticlesCache();
+    List<Article> articlesList;
+
+    public ArticleService() throws IOException, ClassNotFoundException {
+        this.articlesList = initializeArticles();
     }
 
-    public Article getArticleById(String id) {
-        return articlesMap.get(id);
+    public Article getArticleById(int id) {
+        return articlesList.get(id);
```

Comment:
Przeczytaj co zwraca get na liście, list.get(0) -> zwraca obiekt na indexie 0 jak w tablicy a nie obiekt który ma id 0. W mapie jest inaczej bo w mapie jest zwracany klucz o takiej wartości a my wrzuciliśmy obiekty i ich idiki jako klucze więc mogliśmy tak wyciągać. Musisz to poprawić bo to nie będzie działać tak jak chcemy